eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -106,3 +164,12 @@ jobs:
       ctest -C Debug
     workingDirectory: build
     displayName: ctest
+  # Publish Test Results
+  # Publish test results to Azure Pipelines
+  - task: PublishTestResults@2
+    inputs:
+      testResultsFormat: 'JUnit'
+      testResultsFiles: '**/*-test-results.xml' 
+      mergeTestResults: true
+      testRunTitle: 'Azure SDK C99 - $(vm.image)_$(VCPKG_DEFAULT_TRIPLET)'
+    condition: and(succeeded(), contains(variables['vcpkg.deps'], 'cmocka')))
```

Comment:
We always want to upload test results if they're available so this is one of the few exceptions where we shouldn't be checking `succeeded()`

Reply:
I was trying to avoid failing here if no test results files were generated at all